_drafts/oryx/dependency-injection/providing-services.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Providing services
+description: Recommended conventions for defining services
+template: concept-topic-template
+last_updated: Apr 13, 2023
+---
+
+To use and inject services, they must be provided in the DI container. Oryx offers several types of providers:
+
+- `ClassProvider<T>`: Binds a token to a class constructor.
+- `ValueProvider<T>`: Binds a token to a value.
+- `FactoryProvider<T>`: Binds a token to a factory function.
+- `ExistingProvider<T>`: Binds a token to an existing token.
+
+You can pass a list of providers as an argument to the `Injector` constructor or to the Injector's `provide` method.
+
+{% info_block warningBox "When to provide services" %}
+
+Providing services is only possible before the first usage of the DI container.
+
+{% endinfo_block %}
+
+In Oryx applications, providers are typically grouped into const arrays and passed to the app orchestrator either directly or as part of feature definitions. By doing this, you can easily manage features with corresponding providers and their dependencies.
+
+Here's an example describing how to use the app orchestrator to provide services as part of a feature:
+
+```ts
+export const app = appBuilder()
+  // ...
+  .withFeature({
+    providers: [{ provide: CartService, useClass: CustomizedCartService }],
+  })
+  // ...
+  .create();
+```
+
+In this example, the customized cart service is provided by specifying the provider's property within a feature object. This way, you can easily manage and extend the services provided by your Oryx application.
+
+## Injecting the INJECTOR token
```

Comment:
The below part is not technically about providing services, but bit more advanced use cases(both injecting injector and multiproviders). Maybe deserves separate sub chapter?

Reply:
They are already sub-chapters because the main info is under h1

Reply:
Valid point. Can you come up with a name?

Reply:
What about moving first part (above Injecting the Injector token) to the previous chapter ("defining services") and name this one "advanced techniques" or something like this?